### PR TITLE
stm32wl: add register descriptions for dual-core variants

### PR DESCRIPTION
**Pull request by newAM**
This adds register descriptions for registers that are exclusive to the dual-core STM32WL's

cc @jorgeig-space

### CODE REVIEW

File: peripherals/flash/flash_wl.yaml [hunk 1]
```diff
@@ -156,7 +156,7 @@ FLASH:
         Done: [0, "Options modification completed or idle"]
       _write:
         Start: [1, "Trigger options programming operation"]
-    PNB: [0, 0x63]
+    PNB: [0, 0x7F]
```
Comment (newAM) [inline peripherals/flash/flash_wl.yaml:159]: I think this is an error in the STM32WLE reference manual, the STM32WLE manual only goes from page 0 - 63 (the value should be 0x3F in that case), but in the STM32WL5X reference manual this goes from 0 - 127.

As far as I can tell the STM32WLE is capable of programming its whole 256K, same as the STM32WL5X.

File: peripherals/ipcc/ipcc_wl.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      Unmasked: [0, "Transmit channel n free interrupt not masked"]
+    "CH[1-6]OM":
+      Masked: [1, "Receive channel n occupied interrupt masked"]
+      Unmasked: [0, "Receive channel n occupied interrupt not masked"]
+  "C[12]SCR":
+    "CH[1-6]S":
+      Clear: [1, "Processor transmit channel n status bit set"]
+      NoAction: [0, "No action"]
+    "CH[1-6]C":
+      Clear: [1, "Processor receive channel n status bit set"]
```
Comment (adamgreig) [inline peripherals/ipcc/ipcc_wl.yaml:21]:
```suggestion
      Set: [1, "Processor transmit channel n status bit set"]
      NoAction: [0, "No action"]
    "CH[1-6]C":
      Clear: [1, "Processor receive channel n status bit clear"]
```

File: peripherals/flash/flash_wl_c2.yaml [hunk 3]
```diff
@@ -0,0 +1,145 @@
+FLASH:
+  IPCCBR:
+    IPCCDBA: [0, 0x1FFF]
```
Comment (adamgreig) [inline peripherals/flash/flash_wl_c2.yaml:3]:
```suggestion
    IPCCDBA: [0, 0x3FFF]
```

**Review by adamgreig — CHANGES_REQUESTED** [review]
Thanks, this looks good! I just noticed a couple of minor typos, I'll add the fixes for those and get this merged.

**Review by adamgreig — APPROVED** [review]
bors merge

**Pull request merged**